src/System.Private.CoreLib/shared/System/Text/StringBuilder.cs

```diff
@@ -1083,7 +1083,7 @@ public StringBuilder Insert(int index, string? value, int count)
                 {
                     while (count > 0)
                     {
-                        ReplaceInPlaceAtChunk(ref chunk, ref indexInChunk, valuePtr, value.Length);
+                        ReplaceInPlaceAtChunk(ref chunk!, ref indexInChunk, valuePtr, value.Length);
```

Comment:
should you add chunk != null assert instead?

Reply:
Yes, makes sense

Reply:
Actually I can't add a Debug.Assert here because chunk can have a null value after calling `ReplaceInPlaceAtChunk` but the local defined variable is not nullable. Since the target type doesn't match and it is a ref, the compiler is complaining with types not matching because this could potentially take a null value (which at the end of `ReplaceInPlaceAtChunk` it would be null.

Reply:
explained offline, ! is needed here